packages/shared_preferences/shared_preferences/example/integration_test/shared_preferences_test.dart

```diff
@@ -126,11 +126,35 @@ void main() {
       preferences = await SharedPreferences.getInstance();
     });
 
-    tearDown(() {
-      preferences.clear();
+    tearDown(() async {
+      await preferences.clear();
       SharedPreferences.resetStatic();
     });
 
     runAllTests();
   });
+
+  testWidgets('allowList only gets allowed items', (WidgetTester _) async {
+    const Set<String> allowList = <String>{'stringKey', 'boolKey'};
```

Comment:
Consider using constants or final for the multiple usages  of `"stringKey"` and `"boolKey"`. 
Consider naming the variables something that indicates their status. 
```
final String allowedString = 'stringKey';
final String allowedBool = 'boolKey';
final String notAllowedDouble = 'doubleKey';
const Set<String> allowList = <String>{allowedBool, allowedBool};
...
final bool? testBool = prefs.getBool(allowedBool);
expect(testBool, true);

final double? testDouble = prefs.getDouble(notAllowedDouble);
expect(testDouble, null);
```